### PR TITLE
Feature/data/btad

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description

- This PR adds BTec dataset into the repo and refactors MVTec

## Note
It's important to note that furter refactoring could be possible to abstract, but this would be done after all the datasets of interest are added to the repo.

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: anomalib/data/btech.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    samples = pd.DataFrame(samples_list, columns=["path", "split", "label", "image_path"])
+    samples = samples[samples.split != "ground_truth"]
+
+    # Create mask_path column
+    samples["mask_path"] = (
+        samples.path
+        + "/ground_truth/"
+        + samples.label
+        + "/"
+        + samples.image_path.str.rstrip("bmp").str.rstrip(".")
```
Comment (ashwinvaidya17) [inline anomalib/data/btech.py:110]: Can we use `with_suffix` here?

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks for this. I only have one minor comment.

File: anomalib/data/btech.py (hunk 1 above)
Comment (samet-akcay) [reply anomalib/data/btech.py:110]: Yes, we can!

**Review by djdameln — APPROVED** [review]
Thanks!

**Pull request merged**